lib/arm/arm_lifter.ml

```diff
@@ -334,8 +334,12 @@ let lift_bits mem ops (insn : bits_insn ) =
     exec [assn (Env.of_reg dest) rev] cond
   | `REV16, [|`Reg dest; src; cond; _|] ->
     let s = exp_of_op src in
-    let i16 = int32 16 in
-    let rev = Bil.(s lsl i16 lor s lsr i16) in
+    let i8 = int32 8 in
+    let rev = Bil.((s land 0xff lsl i8) lor 
```

Comment:
It will not compile. Also, keep in mind that `s` here is not changing, so you're shifting the same value left or right by 8 bits and never reach higher bits.